lib/controllers/base_controller.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:star_book/controllers/activity.dart';
+
+
+class BaseController{
+  Future<void> initialize();
+  Future<void> dispose();
+
+  Future<void> reset() async{
+    await dispose();
+    await initialize();
+  }
+
+
+  @override
```

Comment:
we do not need to override this method, since you and only calling super here. so that does not make sense.